File: lib/feature/payment/domain/model/enum/payment_category_key.dart
enum PaymentCategoryKey {
  va,
  qris,
  cc,
  ewallet,
  retail,
  unknown;

  static PaymentCategoryKey fromKey(String? key) {
    if (key == null) return PaymentCategoryKey.unknown;
    final lower = key.toLowerCase().trim();
    switch (lower) {
      case 'va':
        return PaymentCategoryKey.va;
      case 'qris':
        return PaymentCategoryKey.qris;
      case 'cc':
        return PaymentCategoryKey.cc;
      case 'ewallet':
        return PaymentCategoryKey.ewallet;
      case 'retail':
        return PaymentCategoryKey.retail;
      default:
        return PaymentCategoryKey.unknown;
    }
  }

  bool get isQris => this == PaymentCategoryKey.qris;
  bool get isVa => this == PaymentCategoryKey.va;
  bool get isCreditCard => this == PaymentCategoryKey.cc;
  bool get isEwallet => this == PaymentCategoryKey.ewallet;
  bool get isRetail => this == PaymentCategoryKey.retail;
}

File: src/features/payment/domain/model/enum/payment_category_key.ts
export enum PaymentCategoryKey {
  VA = "va",
  CC = "cc",
  QRIS = "qris",
  EWALLET = "ewallet",
  RETAIL = "retail",
  OTHER = "other",
}

export const fromCategoryKey = (key?: string | null): PaymentCategoryKey => {
  if (!key) return PaymentCategoryKey.OTHER;
  const k = key.toLowerCase();
  switch (k) {
    case "va":
    case "virtual-account":
    case "virtual_account":
      return PaymentCategoryKey.VA;
    case "cc":
    case "credit-card":
    case "credit_card":
      return PaymentCategoryKey.CC;
    case "qris":
      return PaymentCategoryKey.QRIS;
    case "ewallet":
    case "e-wallet":
      return PaymentCategoryKey.EWALLET;
    case "retail":
    case "convenience-store":
      return PaymentCategoryKey.RETAIL;
    default:
      return PaymentCategoryKey.OTHER;
  }
};
